File: mobile-client/android/app/src/main/java/com/smartresponsor/mobile/app/order/OrderTaxationBridge.kt
package com.smartresponsor.mobile.app.order

import com.smartresponsor.mobile.app.taxation.TaxationFeatureBridge
import com.smartresponsor.mobile.client.contract.taxation.detail.TaxDetailPayload
import com.smartresponsor.mobile.client.contract.taxation.summary.LoadTaxSummaryQuery
import com.smartresponsor.mobile.client.contract.taxation.summary.TaxSummary
import com.smartresponsor.mobile.client.data.taxation.detail.TaxDetailGateway
import com.smartresponsor.mobile.client.data.taxation.summary.TaxSummaryGateway
import com.smartresponsor.mobile.client.usecase.taxation.detail.LoadTaxDetailUseCase
import com.smartresponsor.mobile.client.usecase.taxation.summary.LoadTaxSummaryUseCase

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * Order-owned taxation bridge.
 *
 * Active surface is gateway-first so the order-owned wrapper no longer depends on
 * the flat TaxationFeatureBridge for runtime behavior. A compatibility constructor
 * remains for transitional call sites.
 */
class OrderTaxationBridge(
    private val summaryGateway: TaxSummaryGateway,
    private val detailGateway: TaxDetailGateway,
) {
    constructor(feature: TaxationFeatureBridge) : this(
        summaryGateway = feature.summaryGateway(),
        detailGateway = feature.detailGateway(),
    )

    suspend fun summary(query: LoadTaxSummaryQuery): TaxSummary =
        LoadTaxSummaryUseCase(summaryGateway).invoke(query)

    suspend fun detail(taxationId: String): TaxDetailPayload =
        LoadTaxDetailUseCase(detailGateway).invoke(taxationId)
}

File: client/ios/App/Order/OrderTaxationBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// Order-owned taxation bridge.
//
// Active surface is gateway-first so the order-owned wrapper no longer depends on
// the flat TaxationFeatureBridge for runtime behavior. A compatibility initializer
// remains for transitional call sites.
public struct OrderTaxationBridge: Sendable {
    private let summaryGateway: any TaxSummaryGateway
    private let detailGateway: any TaxDetailGateway

    public init(
        summaryGateway: any TaxSummaryGateway,
        detailGateway: any TaxDetailGateway
    ) {
        self.summaryGateway = summaryGateway
        self.detailGateway = detailGateway
    }

    public init(feature: TaxationFeatureBridge) {
        self.summaryGateway = feature.summaryGatewayRef()
        self.detailGateway = feature.detailGatewayRef()
    }

    public func summary(query: LoadTaxSummaryQuery) async throws -> TaxSummary {
        try await LoadTaxSummaryUseCase(gateway: summaryGateway)(query: query)
    }

    public func detail(taxationId: String) async throws -> TaxDetailPayload {
        try await LoadTaxDetailUseCase(gateway: detailGateway)(taxationId: taxationId)
    }
}
